src/parser/color.rs

```diff
@@ -0,0 +1,641 @@
+                        let a = parse_num(parser, 0.0, 1.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::CloseParenthesis())?;
+                        Ok(Color::Rgb { r, g, b, a })
+                    },
+                    "hsl" => {
+                        parser.optional_whitespace();
+                        let h = parse_num(parser, 0.0, 360.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::Comma())?;
+                        parser.optional_whitespace();
+                        let s = parse_percent(parser, 0.0, 100.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::Comma())?;
+                        parser.optional_whitespace();
+                        let l = parse_percent(parser, 0.0, 100.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::CloseParenthesis())?;
+                        Ok(Color::Hsl { h, s, l, a: 1.0 })
+                    },
+                    "hsla" => {
+                        parser.optional_whitespace();
+                        let h = parse_num(parser, 0.0, 360.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::Comma())?;
+                        parser.optional_whitespace();
+                        let s = parse_percent(parser, 0.0, 100.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::Comma())?;
+                        parser.optional_whitespace();
+                        let l = parse_percent(parser, 0.0, 100.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::Comma())?;
+                        parser.optional_whitespace();
+                        let a = parse_num(parser, 0.0, 1.0)?;
+                        parser.optional_whitespace();
+                        parser.expect(Token::CloseParenthesis())?;
+                        Ok(Color::Hsl { h, s, l, a })
+                    },
+                    _ => Err(ParsingError::wrong_token(token_at, "rgb, rgba, hsl, or hsla")),
+                },
+                Token::Hash(value, _) => {
+                     match value.len() {
+                        3 => {
+                            let mut red_string = String::new();
+                            let mut green_string = String::new();
+                            let mut blue_string = String::new();
+                            red_string.push_str(&value[0..1]);
+                            red_string.push_str(&value[0..1]);
+                            green_string.push_str(&value[1..2]);
+                            green_string.push_str(&value[1..2]);
+                            blue_string.push_str(&value[2..3]);
+                            blue_string.push_str(&value[2..3]);
```

Comment:
It would probably be better to make a helper function for this but I guess this is fine

Reply:
I did this in e1866e7